MmSupervisorPkg/Core/Mem/SmmCpuMemoryManagement.c

```diff
@@ -230,18 +230,29 @@ InitializePageTablePool (
   returned.
 
   @param  Pages                 The number of 4 KB pages to allocate.
+  @param  NewAllocation         Pointer to a passed in BOOLEAN that will be TRUE if new pages have been allocated
+                                for the page pool and FALSE otherwise.
 
   @return A pointer to the allocated buffer or NULL if allocation fails.
 
 **/
 VOID *
 AllocatePageTableMemory (
-  IN UINTN  Pages
+  IN UINTN     Pages,
+  OUT BOOLEAN  *NewAllocation OPTIONAL
   )
 {
   VOID  *Buffer;
 
+  if (NewAllocation != NULL) {
+    *NewAllocation = TRUE;
+  }
+
   if (Pages == 0) {
+    if (NewAllocation != NULL) {
+      *NewAllocation = FALSE;
+    }
+
     return NULL;
   }

```

Comment:
This is can be a little more concise.

```suggestion
  if (NewAllocation != NULL) {
    if (Pages == 0) {
      *NewAllocation = FALSE;
      return NULL;
    }
    *NewAllocation = TRUE;
  }
```